src/main/cpp/DragonVision/DragonPhotonCam.cpp

```diff
@@ -28,21 +37,344 @@ DragonPhotonCam::DragonPhotonCam(std::string name,
+
+        // Get transformation from camera to target
+        frc::Transform3d transform = target.GetBestCameraToTarget();
+        // just need to add translation components of transforms together (camToTarget.X() + robotToCam.X())
+        return target.GetBestCameraToTarget().X() + m_robotCenterToCam.X();
+    }
+
+    return units::length::inch_t(-1.0);
+}
+
+/// @brief Estimate the Y distance to the detected target in relation to robot
+/// @return units::length::inch_t - Positive is left
+units::length::inch_t DragonPhotonCam::EstimateTargetYDistance_RelToRobotCoords() const
+{
+    ///@TODO: May have problems when Multi-tag is enabled, data may not come through
+
+    // get latest detections
+    photon::PhotonPipelineResult result = m_camera->GetLatestResult();
+
+    // check for detections
+    if (result.HasTargets())
+    {
+        // get the most accurate data according to contour ranking
+        photon::PhotonTrackedTarget target = result.GetBestTarget();
+
+        // Get transformation from camera to target
+        frc::Transform3d transform = target.GetBestCameraToTarget();
+        // just need to add translation components of transforms together (camToTarget.X() + robotToCam.X())
+        return target.GetBestCameraToTarget().Y() + m_robotCenterToCam.Y();
+    }
+
+    return units::length::inch_t(-1.0);
+}
+
+/// @brief Estimate the Z distance to the detected target in relation to robot
+/// @return units::length::inch_t - Positive is up
+
+units::length::inch_t DragonPhotonCam::EstimateTargetZDistance_RelToRobotCoords() const
+{
+    ///@TODO: May have problems when Multi-tag is enabled, data may not come through
+
+    // get latest detections
+    photon::PhotonPipelineResult result = m_camera->GetLatestResult();
+
+    // check for detections
+    if (result.HasTargets())
+    {
+        // get the most accurate data according to contour ranking
+        photon::PhotonTrackedTarget target = result.GetBestTarget();
+
+        // Get transformation from camera to target
+        frc::Transform3d transform = target.GetBestCameraToTarget();
+
+        // just need to add translation components of transforms together (camToTarget.X() + robotToCam.X())
+        return target.GetBestCameraToTarget().Z() + m_robotCenterToCam.Z();
+    }
+
+    return units::length::inch_t(-1.0);
+}
 bool DragonPhotonCam::SetPipeline(DragonCamera::PIPELINE pipeline)
 {
     return false;
+}
+
+VisionData DragonPhotonCam::GetDataToNearestAprilTag()
+{
+    // get latest detections from co-processor
+    frc::Transform3d camToTargetTransform;
+    photon::PhotonPipelineResult result = m_camera->GetLatestResult();
+    if (result.HasTargets())
+    {
+        // get the most accurate according to configured contour ranking
+        photon::PhotonTrackedTarget target = result.GetBestTarget();
+
+        camToTargetTransform = target.GetBestCameraToTarget();
+    }
+    return VisionData{camToTargetTransform.Translation()};
```

Comment:
Can return AprilTagId by calling result.GetFiducialId? some function similar to that name